### ISSUE TITLE
P_Mapping_failed in version 2.3 with HGAP3 protocol

### ISSUE THREAD

**Issue by sagarutturkar** [PacificBiosciences/SMRT-Analysis#164]
Hi,

I recently installed SMRT Analysis version 2.3 and started a HGAP3 job for Bacterial genome. However it failed with P_mapping. Can you help with this.

The link for the log file is:

https://gist.github.com/sagarutturkar/a3f374fc379b2711dc10

Thanks
Sagar

**Comment by sagarutturkar** [comment]
The align.log file has following message:

echo "Finished on $(date -u)"
echo "Task align with nproc 47 with exit code ${rcode}."
exit ${rcode}# Writing stdout and stderr from Popen:
Setting up ENV on mamba for task align
Running task://016440/P_Mapping/align on Linux mamba 3.5.0-45-generic #68~precise1-Ubuntu SMP Wed Dec 4 16:18:46 UTC 2013 x86_64 x86_64 x86_64 GNU/Linux
Started on Mon Jan  5 19:40:47 UTC 2015
Validating existence of Input Files
Successfully found /data2/smart/smrtanalysis/userdata/jobs/016/016440/reference
Successfully found /data2/smart/smrtanalysis/userdata/jobs/016/016440/input.fofn
Successfully found /data2/smart/smrtanalysis/userdata/jobs/016/016440/data/filtered_regions.fofn
Successfully validated input files
Task 1 completed at Mon Jan  5 15:19:32 EST 2015
Alignment Complete
Task 2 completed at Mon Jan  5 15:19:32 EST 2015
Traceback (most recent call last):
  File "/data2/smart/smrtanalysis/install/smrtanalysis_2.3.0.140936/analysis/bin/loadChemistry.py", line 8, in <module>
    load_entry_point('pbalign==0.2.0', 'console_scripts', 'loadChemistry.py')()
  File "/data2/smart/smrtanalysis/install/smrtanalysis_2.3.0.140936/redist/python2.7/lib/python2.7/site-packages/pbalign/tools/loadChemistry.py", line 72, in main
    chemTriple = bas.chemistryBarcodeTriple
  File "/data2/smart/smrtanalysis/install/smrtanalysis_2.3.0.140936/redist/python2.7/lib/python2.7/site-packages/pbcore/io/BasH5IO.py", line 860, in chemistryBarcodeTriple
    return self._parts[0].chemistryBarcodeTriple
  File "/data2/smart/smrtanalysis/install/smrtanalysis_2.3.0.140936/redist/python2.7/lib/python2.7/site-packages/pbcore/io/BasH5IO.py", line 624, in chemistryBarcodeTriple
    raise ChemistryLookupError, "Could not find chemistry barcodes in file or companion metadata.xml"
pbcore.chemistry.chemistry.ChemistryLookupError: Could not find chemistry barcodes in file or companion metadata.xml

**Comment by mjhsieh** [comment]
Do you have the corresponding metadata.xml file(s) along with the smrt cell data?

**Comment by mjhsieh** [comment]
A smrtcell data folder structure is described here: https://github.com/PacificBiosciences/SMRT-Analysis/wiki/Cannot-import-new-SMRT-Cells

**Comment by sagarutturkar** [comment]
Yes. Metadata.xml file is present. The folder structure is correct. I was able to run the HGAp or any other job correctly with this data in v2.2. But Since we installed 2.3, its giving errors.

Data looks like:
MY_Microbe
    XSBOR_20130726_RS42150_PL100036291-1_C02.metadata.xml
    XSBOR_20130726_RS42150_PL100036291-1_D02.metadata.xml
    Analysis_Results
         XSBOR_20130726_RS42150_PL100036291-1_C02.1.bas.h5
         XSBOR_20130726_RS42150_PL100036291-1_D02.1.bas.h5
         XSBOR_20130726_RS42150_PL100036291-1_C02.1.1.bax.h5
         XSBOR_20130726_RS42150_PL100036291-1_C02.1.2.bax.h5
         XSBOR_20130726_RS42150_PL100036291-1_C02.1.3.bax.h5
         XSBOR_20130726_RS42150_PL100036291-1_D02.1.1.bax.h5
         XSBOR_20130726_RS42150_PL100036291-1_D02.1.2.bax.h5
         XSBOR_20130726_RS42150_PL100036291-1_D02.1.3.bax.h5

**Comment by sagarutturkar** [comment]
I was not able to figure out solution for this issue. I reverted back to version 2.2 and its working fine with same data.

**Comment by xiehaiy** [comment]
Do you solve this issue? I meet this issue too

**Comment by sagarutturkar** [comment]
Hi,

I updated to latest version as well changed the names for my data files as
shown below:
https://github.com/PacificBiosciences/SMRT-Analysis/wiki/Cannot-import-new-SMRT-Cells

It worked after this for twi SMRT cells. But Issue was random and I was not
able to use the data from another SMRT cell even after name changing.

Thanks
Sagar

**Comment by piatekmj** [comment]
I can confirm that downgrading the pipeline from v2.3 to v2.2 solves this issue.

It seems the latest software can't handle datasets from all instruments but instead supports only a selection of instruments. That creates a problem for whoever would like to reanalyze older datasets as there is no clear guideline what version of pipeline to use with what version of instrument.

**Comment by mjhsieh** [comment]
Can you list the instrument combination that you think should work in your case?

**Comment by piatekmj** [comment]
I don't think I have a great list to share but so far I found out that with older chemistries, like from RS I, I can only succeed with smrtanalysis v2.2. I have tested few runs only.

From [one of your posts](http://j.mp/2gAmrss) (last accessed at the time of writing this post) it seems that v3.0 is not backward compatible. I'm looking forward to test it once any data from Sequel becomes available.

RS II system seems to be working fine with smrtanalysis v2.3 (although v2.3 still has some unexplained glitches). I have successfully assembled hundreds of genomes with PacBio data.
